File: 03-array/12-leaders-in-an-array-problem/12-leaders-in-an-array-problem.ts
const leaders_in_an_array = (nums: number[]): number[] => {
  const leaders = []
  let last_max = -Infinity
  for (let i = nums.length - 1; i >= 0; i--) {
    const num = nums[i]
    if (num > last_max) {
      leaders.unshift(num)
      last_max = num
    }
  }
  return leaders
}

console.log(leaders_in_an_array([7, 10, 4, 10, 6, 5, 2]))

File: 03-array/12-leaders-in-an-array-problem/12-leaders-in-an-array-problem.rb
def leaders_in_an_array(nums)
  leaders = []
  last_max = -Float::INFINITY
  nums.reverse_each do |num|
    if last_max < num
      leaders.unshift(num)
      last_max = num
    end
  end
  leaders
end

print leaders_in_an_array [7,10,4,10,6,5,2]

